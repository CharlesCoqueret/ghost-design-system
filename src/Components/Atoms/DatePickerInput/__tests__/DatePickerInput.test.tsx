import React from 'react';
import { DatePickerInput } from '..';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('DatePickerInput Component', () => {
  it('DatePickerInput renders handles changes', async () => {
    const onChangeMock = jest.fn();

    render(<DatePickerInput onChange={onChangeMock} name='NAME' placeholder='TESTPLACEHOLDER' />);

    const inputNode = await screen.findByPlaceholderText('TESTPLACEHOLDER');

    userEvent.click(inputNode);
    userEvent.type(inputNode, '3/2/22{enter}');

    // Using dynamic date instead of static value to ensure local time is taken into consideration.
    const expectedDate = new Date('Wed Mar 02 2022');

    await waitFor(() => {
      expect(onChangeMock).toHaveBeenLastCalledWith(expectedDate, expect.anything());
    });
  });

  it('DatePickerInput renders and changes without onChange prop', async () => {
    render(<DatePickerInput name='NAME' placeholder='TESTPLACEHOLDER' />);

    const inputNode = await screen.findByPlaceholderText('TESTPLACEHOLDER');

    userEvent.click(inputNode);
    userEvent.type(inputNode, '3/2/22{enter}');

    await waitFor(() => {
      expect(inputNode).not.toBeNull();
    });
  });

  it('DatePickerInput renders highlighted in readonly', async () => {
    const { container } = render(<DatePickerInput name='NAME' highlighted readOnly dateFormat={''} />);
    expect(container).toMatchSnapshot();
  });

  it('DatePickerInput renders with placeholder', async () => {
    const { container } = render(<DatePickerInput name='NAME' placeholder='TESTPLACEHOLDER' />);
    expect(container).toMatchSnapshot();
  });

  it('DatePickerInput renders without Portal', async () => {
    const { container } = render(<DatePickerInput usePortal={false} name='NAME' />);
    expect(container).toMatchSnapshot();
  });

  it('DatePickerInput renders highlighted in disabled', async () => {
    const { container } = render(<DatePickerInput name='NAME' highlighted disabled />);
    expect(container).toMatchSnapshot();
  });

  it('DatePickerInput renders in error in readonly', async () => {
    const { container } = render(<DatePickerInput name='NAME' readOnly isInError />);
    expect(container).toMatchSnapshot();
  });

  it('DatePickerInput renders in error', async () => {
    const { container } = render(<DatePickerInput name='NAME' isInError />);
    expect(container).toMatchSnapshot();
  });
});
