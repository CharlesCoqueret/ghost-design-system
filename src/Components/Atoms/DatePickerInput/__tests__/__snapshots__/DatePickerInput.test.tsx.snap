// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerInput Component DatePickerInput renders and changes without onChange prop 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input"
          placeholder="TESTPLACEHOLDER"
          tabindex="0"
          type="text"
          value=""
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`DatePickerInput Component DatePickerInput renders and changes without onChange prop 2`] = `
<div>
  <div
    class=""
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input"
          placeholder="TESTPLACEHOLDER"
          tabindex="0"
          type="text"
          value=""
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`DatePickerInput Component DatePickerInput renders highlighted in disabled 1`] = `
<div>
  <div
    class="input-date-picker-wrapper-read-only"
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input-read-only input-date-picker-input highlighted"
          disabled=""
          placeholder="-"
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`DatePickerInput Component DatePickerInput renders highlighted in readonly 1`] = `
<div>
  <div
    class="input-date-picker-wrapper-read-only"
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input-read-only highlighted"
          placeholder="-"
          readonly=""
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`DatePickerInput Component DatePickerInput renders in error 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input input-error"
          placeholder="MMM DD, YYYY"
          tabindex="0"
          type="text"
          value=""
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`DatePickerInput Component DatePickerInput renders in error in readonly 1`] = `
<div>
  <div
    class="input-date-picker-wrapper-read-only"
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input-read-only"
          placeholder="-"
          readonly=""
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`DatePickerInput Component DatePickerInput renders with placeholder 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input"
          placeholder="TESTPLACEHOLDER"
          tabindex="0"
          type="text"
          value=""
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`DatePickerInput Component DatePickerInput renders without Portal 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="input-date-picker-input"
          placeholder="MMM DD, YYYY"
          tabindex="0"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
